src/app/routes/connector-ui/transfer-history-page/transfer-history-page/transfer-history-page.component.ts

```diff
@@ -20,70 +25,83 @@ import {Fetched} from '../../../../core/services/models/fetched';
 })
 export class TransferHistoryPageComponent implements OnInit {
   columns: string[] = [
-    'id',
-    'creationDate',
+    'direction',
+    'assetId',
     'state',
     'lastUpdated',
-    'connectorId',
-    'assetId',
-    'contractId',
+    'counterPartyConnectorEndpoint',
     'details',
   ];
   transferProcessesList: Fetched<{
-    transferProcesses: Array<TransferProcessDto>;
+    transferProcesses: Array<TransferHistoryEntry>;
   }> = Fetched.empty();
 
   constructor(
-    private transferProcessService: TransferProcessService,
+    private edcApiService: EdcApiService,
+    private assetDetailDialogDataService: AssetDetailDialogDataService,
+    private assetPropertyMapper: AssetPropertyMapper,
+    private notificationService: NotificationService,
     private dialog: MatDialog,
-  ) {}
+  ) {
+  }
 
-  onTransferHistoryDetailsClick(item: TransferProcessDto) {
+  onTransferHistoryDetailsClick(item: TransferHistoryEntry) {
     const data: JsonDialogData = {
-      title: item.id,
+      title: item.assetName ?? item.assetId,
       subtitle: 'Transfer History Details',
       icon: 'assignment',
       objectForJson: item,
     };
     this.dialog.open(JsonDialogComponent, {data});
   }
 
-  ngOnInit(): void {
-    this.loadTransferProcesses();
+  loadAssetDetails(transferProcessId: string): Observable<Fetched<Asset>> {
+    return this.edcApiService.getTransferProcessAsset(transferProcessId)
+      .pipe(
+        map((asset) =>
+          this.assetPropertyMapper.buildAssetFromProperties(asset.properties),
+        ),
+        Fetched.wrap({
```

Comment:
no need for fetched.wrap, just subscribe to the result, if it's a success, you show the dialog, if it's a failure, you show an alert "failed fetching bla bli blub" and log the error.